In [49]:
import pandas 
import dimcli
import plotly
import plotly.graph_objs as go
from plotly.offline import download_plotlyjs, init_notebook_mode, plot
init_notebook_mode(connected=True)
dimcli.login()

DimCli v0.4.8 - Succesfully connected to <https://app.dimensions.ai> (method: dsl.ini file)


HOWTO Install the dimensions client library: https://nbviewer.jupyter.org/github/digital-science/dimensions-api/blob/master/1.Getting%20Started/1.%20DimCli%20library%20for%20Dimensions.ipynb

## Publications of Heidelberg university with at least one author with a non-German affiliation

Heidelberg University is https://grid.ac/institutes/grid.7700.0 on GRID.

In [2]:
%%dsl_query
search publications where research_orgs="grid.7700.0" and count(research_org_countries) > 1 limit 1

Tot Publications:  31706
Returned Publications: 20


<dimcli.Result object #4531103224: [('publications', 20), ('_stats', 1)]>

**31706** publications in total with international collaborators.

### Distribution over years

In [18]:
%%dsl_query
search publications where research_orgs="grid.7700.0" and count(research_org_countries) > 1 return year

Tot Publications:  31715
Returned Year: 20


<dimcli.Result object #4777619352: [('year', 20), ('_stats', 1)]>

In [19]:
df = dsl_last_results.as_dataframe()
df

count    id
0    2751  2018
1    2349  2017
2    2163  2016
3    2018  2014
4    2013  2015
5    1944  2012
6    1914  2013
7    1528  2011
8    1421  2010
9    1291  2019
10   1264  2009
11   1112  2008
12   1063  2007
13    932  2006
14    712  2005
15    688  2004
16    564  2002
17    556  2003
18    437  2001
19    416  2000

In [20]:
data = [go.Bar(
            x=[x for x in df['id']],
            y=[x for x in df['count']]
    )]
plot(data, filename="years_distribution.html")

'years_distribution.html'

### List of collaborating institutions with count

In [32]:
%%dsl_query
search publications where research_orgs="grid.7700.0" and count(research_org_countries) > 1 return research_orgs limit 100

Tot Publications:  31715
Returned Research_orgs: 100


<dimcli.Result object #4808838616: [('research_orgs', 100), ('_stats', 1)]>

In [37]:
df = dsl_last_results.as_dataframe()
df

acronym  count    country_name             id  \
0       NaN  31715         Germany    grid.7700.0   
1      CERN   2578     Switzerland    grid.9132.9   
2       RAL   2559  United Kingdom  grid.76978.37   
3       NaN   2521  United Kingdom    grid.5379.8   
4       NaN   2476  United Kingdom    grid.6572.6   
5       NaN   2320  United Kingdom  grid.10025.36   
6       NaN   2282  United Kingdom    grid.4305.2   
7       NaN   2252  United Kingdom    grid.5335.0   
8       LNF   2215           Italy  grid.463190.9   
9       LAL   2202          France  grid.462450.1   
10      NaN   2061         Germany    grid.5675.1   
11     BINP   1991          Russia  grid.418495.5   
12   NIKHEF   1989     Netherlands  grid.420012.5   
13    LPNHE   1973          France  grid.463935.e   
14      NaN   1946   United States  grid.38142.3c   
15      NaN   1937  United Kingdom    grid.4991.5   
16     LAPP   1906          France  grid.450330.1   
17     ITEP   1899          Russia  grid.21626.31   
18      OSU   1847   United States  grid.261331.4   
19      UCL   1810  United Kingdom  grid.83440.3b   
20      NaN   1787          Poland  grid.418860.3   
21      LMU   1771         Germany    grid.5252.0   
22      NaN   1769  United Kingdom    grid.8756.c   
23      NaN   1742          Canada  grid.14848.31   
24      NaN   1739          Norway    grid.7914.b   
25      NaN   1729         Germany  grid.10388.32   
26      NaN   1720  United Kingdom    grid.7445.2   
27      NaN   1701   United States  grid.47100.32   
28      UBC   1690          Canada  grid.17091.3e   
29      NaN   1687  United Kingdom    grid.7372.1   
..      ...    ...             ...            ...   
70       HU   1333         Germany    grid.7468.d   
71     UFRJ   1329          Brazil    grid.8536.8   
72      NaN   1321           Italy  grid.470206.7   
73  AGH UST   1317          Poland    grid.9922.0   
74      UTD   1295   United States  grid.267323.1   
75      MPP   1294         Germany  grid.435824.c   
76      NaN   1285   United States  grid.25879.31   
77      NaN   1281  United Kingdom    grid.9835.7   
78      NaN   1280          Israel  grid.13992.30   
79      NaN   1261          Greece    grid.5216.0   
80      NaN   1255          Canada  grid.17063.33   
81      USP   1252          Brazil  grid.11899.38   
82     AANL   1252         Armenia  grid.48507.3e   
83     IHEP   1250           China  grid.418741.f   
84      NaN   1241           Italy  grid.470219.9   
85       UB   1236     Switzerland    grid.5734.5   
86     UMCP   1226   United States  grid.164295.d   
87      NaN   1212         Czechia    grid.4491.8   
88      NaN   1206   United States  grid.89336.37   
89      NaN   1204       Australia    grid.1008.9   
90      NaN   1198         Germany    grid.8379.5   
91  IEP SAS   1194        Slovakia  grid.435184.f   
92      FZU   1193         Czechia  grid.424881.3   
93      NaN   1186         Austria    grid.5771.4   
94      NaN   1182  United Kingdom  grid.11835.3e   
95       UW   1173   United States  grid.34477.33   
96      GAU   1158         Germany    grid.7450.6   
97      NaN   1154           Italy  grid.470190.b   
98      UiO   1147          Norway    grid.5510.1   
99      NaN   1140          Israel    grid.6451.6   

                                                 name  
0                               Heidelberg University  
1          European Organization for Nuclear Research  
2                      Rutherford Appleton Laboratory  
3                            University of Manchester  
4                            University of Birmingham  
5                             University of Liverpool  
6                             University of Edinburgh  
7                             University of Cambridge  
8                     National Laboratory of Frascati  
9              Laboratoire de l'Accélérateur Linéaire  
10                             TU Dortmund University  
11                Budker Institute of Nuclear

In [38]:
df = df.iloc[1:] # remove first row (= Heidelberg)

In [39]:
data = [go.Bar(
            x=[x for x in df['name']],
            y=[x for x in df['count']]
    )]
plot(data, filename="org_distribution.html")

'org_distribution.html'

### List of collaborating countries with count

In [40]:
%%dsl_query
search publications where research_orgs="grid.7700.0" and count(research_org_countries) > 1 return research_org_countries limit 100

Tot Publications:  31715
Returned Research_org_countries: 100


<dimcli.Result object #4823003992: [('research_org_countries', 100), ('_stats', 1)]>

In [41]:
df = dsl_last_results.as_dataframe()
df

count  id                  name
0   31715  DE               Germany
1   12793  US         United States
2    8716  GB        United Kingdom
3    6914  FR                France
4    6585  CH           Switzerland
5    5982  IT                 Italy
6    5014  NL           Netherlands
7    4195  ES                 Spain
8    3880  CN                 China
9    3717  CA                Canada
10   3644  AT               Austria
11   3547  RU                Russia
12   3350  SE                Sweden
13   3037  JP                 Japan
14   2923  PL                Poland
15   2741  AU             Australia
16   2405  NO                Norway
17   2337  DK               Denmark
18   2245  BR                Brazil
19   2161  IL                Israel
20   1896  CZ               Czechia
21   1892  GR                Greece
22   1882  RO               Romania
23   1650  BE               Belgium
24   1597  ZA          South Africa
25   1483  TR                Turkey
26   1383  PT              Portugal
27   1342  SK              Slovakia
28   1274  AM               Armenia
29   1268  HU               Hungary
..    ...  ..                   ...
70     92  GH                 Ghana
71     92  TZ              Tanzania
72     85  NA               Namibia
73     83  MN              Mongolia
74     82  KE                 Kenya
75     79  CY                Cyprus
76     74  AE  United Arab Emirates
77     67  BD            Bangladesh
78     60  LV                Latvia
79     58  NG               Nigeria
80     58  PS    State of Palestine
81     53  LB               Lebanon
82     45  UY               Uruguay
83     43  MK             Macedonia
84     38  UG                Uganda
85     37  EC               Ecuador
86     37  KZ            Kazakhstan
87     37  MW                Malawi
88     36  JO                Jordan
89     34  DZ               Algeria
90     34  JM               Jamaica
91     33  LK             Sri Lanka
92     30  CR            Costa Rica
93     27  VE             Venezuela
94     26  HN              Honduras
95     26  ZM                Zambia
96     25  MT                 Malta
97     23  CM              Cameroon
98     22  KW                Kuwait
99     22  TN               Tunisia

[100 rows x 3 columns]

In [43]:
df = df.iloc[1:] # remove first row (= Germany)

In [45]:
data = [go.Bar(
            x=[x for x in df['name']],
            y=[x for x in df['count']]
    )]
plot(data, filename="country_distribution.html")

'country_distribution.html'

### Distribution over FOR codes

In [46]:
%%dsl_query
search publications where research_orgs="grid.7700.0" and count(research_org_countries) > 1 return FOR limit 50

Tot Publications:  31715
Returned For: 50


<dimcli.Result object #4815462976: [('FOR', 50), ('_stats', 1)]>

In [47]:
df = dsl_last_results.as_dataframe()
df

count    id                                               name
0    3100  3053                             1103 Clinical Sciences
1    3029  2581                 0601 Biochemistry and Cell Biology
2    2075  2421                       0299 Other Physical Sciences
3    2037  3177             1117 Public Health and Health Services
4    1980  2389  0202 Atomic, Molecular, Nuclear, Particle and ...
5    1790  2620                                      0604 Genetics
6    1684  3120                                 1109 Neurosciences
7    1512  3142                   1112 Oncology and Carcinogenesis
8    1150  2471         0306 Physical Chemistry (incl. Structural)
9    1115  3048    1102 Cardiorespiratory Medicine and Haematology
10   1051  2377               0201 Astronomical and Space Sciences
11    796  3468                                    1701 Psychology
12    775  2746  0801 Artificial Intelligence and Image Processing
13    659  3103                                    1107 Immunology
14    628  2358                                    0104 Statistics
15    573  2509                                       0403 Geology
16    528  3114                          1108 Medical Microbiology
17    452  2330                              0101 Pure Mathematics
18    398  3153                   1113 Ophthalmology and Optometry
19    338  2921                         0912 Materials Engineering
20    300  2953                 0915 Interdisciplinary Engineering
21    289  3292                             1402 Applied Economics
22    253  2539  0406 Physical Geography and Environmental Geos...
23    232  2790                           0806 Information Systems
24    227  2438                           0302 Inorganic Chemistry
25    224  2648                                 0607 Plant Biology
26    204  2503                                  0402 Geochemistry
27    188  3158         1114 Paediatrics and Reproductive Medicine
28    180  2486                       0399 Other Chemical Sciences
29    169  2344                           0102 Applied Mathematics
30    157  3164      1115 Pharmacology and Pharmaceutical Sciences
31    154  2480       0307 Theoretical and Computational Chemistry
32    142  2597                                       0602 Ecology
33    132  2447        0303 Macromolecular and Materials Chemistry
34    126  2464                             0305 Organic Chemistry
35    124  2415                               0206 Quantum Physics
36    124  2837                        0903 Biomedical Engineering
37    123  2878                     0907 Environmental Engineering
38    123  3172                            1116 Medical Physiology
39    119  2428                          0301 Analytical Chemistry
40    118  2558          0502 Environmental Science and Management
41    110  2353       0103 Numerical and Computational Mathematics
42    105  2534                                  0405 Oceanography
43    100  2867         0906 Electrical and Electronic Engineering
44     94  2409                               0205 Optical Physics
45     88  3432                             1606 Political Science
46     81  2891                          0909 Geomatic Engineering
47     80  3675                            2103 Historical Studies
48     78  2766                             0803 Computer Software
49     65  2634                                  0605 Microbiology

In [48]:
data = [go.Pie(
            labels=[x for x in df['name']],
            values=[x for x in df['count']]
    )]
plot(data, filename="for_distribution.html")

'for_distribution.html'